io/zenoh-transport/src/multicast/rx.rs

```diff
@@ -58,197 +68,249 @@ impl TransportMulticastInner {
-        //                             "Transport {}: {}. Unknown conduit {:?} from {}.",
-        //                             self.manager.config.zid,
-        //                             self.locator,
-        //                             channel.priority,
-        //                             peer.locator
-        //                         );
-        //                     };
-
-        //                     let guard = match channel.reliability {
-        //                         Reliability::Reliable => zlock!(c.reliable),
-        //                         Reliability::BestEffort => zlock!(c.best_effort),
-        //                     };
-        //                     self.handle_frame(sn, payload, guard, peer)
-        //                 }
-        //                 TransportBody::Join(join) => self.handle_join_from_peer(join, peer),
-        //                 TransportBody::Close(close) => {
-        //                     drop(r_guard);
-        //                     self.del_peer(locator, close.reason)
-        //                 }
-        //                 _ => Ok(()),
-        //             }
-        //         }
-        //         None => {
-        //             drop(r_guard);
-        //             match msg.body {
-        //                 TransportBody::Join(join) => self.handle_join_from_unknown(join, locator),
-        //                 _ => Ok(()),
-        //             }
-        //         }
-        //     }
-        // }
-        todo!();
+        // Check if parameters are ok
+        if join.version != peer.version
+            || join.zid != peer.zid
+            || join.whatami != peer.whatami
+            || join.resolution != peer.resolution
+            || join.lease != peer.lease
+            || join.ext_qos.is_some() != peer.is_qos()
+        {
+            let e = format!(
+                "Ingoring Join on {} of peer: {}. Inconsistent parameters.",
+                peer.locator, peer.zid,
+            );
+            log::debug!("{}", e);
+            bail!("{}", e);
+        }
+
+        Ok(())
+    }
+
+    pub(super) fn handle_join_from_unknown(&self, join: Join, locator: &Locator) -> ZResult<()> {
+        if zread!(self.peers).len() >= self.manager.config.multicast.max_sessions {
+            log::debug!(
+                "Ingoring Join on {} from peer: {}. Max sessions reached: {}.",
+                locator,
+                join.zid,
+                self.manager.config.multicast.max_sessions,
+            );
+            return Ok(());
+        }
+
+        if join.version != self.manager.config.version {
+            log::debug!(
+                "Ingoring Join on {} from peer: {}. Unsupported version: {}. Expected: {}.",
+                locator,
+                join.zid,
+                join.version,
+                self.manager.config.version,
+            );
+            return Ok(());
+        }
+
+        if join.resolution > self.manager.config.resolution {
```

Comment:
Pico doesn't support resolutions different from its own, wouldn't this mean that a Pico with resolution < manager.config.resoluction would be able to join, but couldn't be joined by this?

Reply:
I believe if the resolution is smaller it should be able to join. The other way around is not true. So, I believe zenoh-pico should support the case when a smaller resolution is advertised.

Reply:
Yeah, but then the sn would need to respect that resolution, which means that every peer on the multicast network must be aware of the minimal resolution, including if it changes as a new peer joins the network. Sounds general's paradoxey to me :/

Reply:
You are right, while on the reading side it would pose no problem, it does on the writing side. So you are correct. In unicast we can do that check and react accordingly with sn resolution, but in multicast is cumbersome. Let's stick to strict equality check for now.